server/src/presentation/socket/handlers/leaveRoom.ts

```diff
@@ -0,0 +1,64 @@
+import { QuasmComponent } from '@quasm/common';
+import { UUID } from 'crypto';
+
+import { Chat } from '@/domain/game/chat/Chat';
+import { logger } from '@/infrastructure/logger/Logger';
+
+import { HandlerConfig } from './HandlerConfig';
+import { withErrorHandling } from './withErrorHandling';
+
+export function leaveRoomHandler({
+    dataAccess,
+    authProvider,
+    socket,
+    io
+}: HandlerConfig) {
+    socket.on('leaveRoom', (roomID, respond) => {
+        withErrorHandling(respond, async () => {
+            authProvider;
+
+            logger.info(
+                QuasmComponent.SOCKET,
+                `${socket.data.userID} | SOCKET leaveRoom RECEIVED ${roomID} `
+            );
+
+            const room = await dataAccess.roomRepository.getRoomByID(
+                roomID as UUID
+            );
+
+            const character = await dataAccess.characterRepository.getCharacter(
```

Comment:
Instead of getCharacter on repo just call `room.getCharacterByUserID` it will give you already existing instance corelated with room so if you update it changes will be reflected everywhere else